src/sinks/util/batch.rs

```diff
@@ -1,7 +1,35 @@
 use futures::{try_ready, Async, AsyncSink, Future, Poll, Sink, StartSend};
+use serde::{Deserialize, Serialize};
 use std::time::{Duration, Instant};
 use tokio::timer::Delay;
 
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct BatchConfig {
+    pub size: Option<usize>,
+    pub timeout: Option<u64>,
+}
+
+pub struct BatchSettings {
+    pub size: usize,
+    pub timeout: Duration,
+}
+
+impl BatchSettings {
+    pub fn from_option(config: &Option<BatchConfig>, size: u64, timeout: u64) -> Self {
+        let size: usize = size as usize;
+        config
+            .as_ref()
+            .map(|c| Self {
+                size: c.size.unwrap_or(size),
+                timeout: Duration::from_secs(c.timeout.unwrap_or(timeout)),
+            })
+            .unwrap_or(Self {
+                size,
+                timeout: Duration::from_secs(timeout),
+            })
```

Comment:
If `config` becomes `&BatchConfig` instead of `&Option<BatchConfig>`, this can probably get much simpler. 

```rust
Self {
    size: config.size.unwrap_or(size),
    timeout: Duration::from_secs(config.timeout.unwrap_or(timeout)),
})
```

Reply:
My original design had it as a method on `BatchConfig` itself, which is made impossible by it being an `Option`. Many possibilites for simplification, though, if it doesn't have to be an `Option`.